aten/src/ATen/functorch/Interpreter.cpp

```diff
@@ -88,10 +89,10 @@ void sanityCheckStack(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
   auto num_args = op.schema().arguments().size();
   foreachTensorInplace(*stack, stack->size() - num_args, stack->size(),
       [](const Tensor& tensor) {
-
-        auto* wrapper = maybeGetTensorWrapper(tensor);
+        auto result = unwrapIfDead(tensor);
+        auto* wrapper = maybeGetTensorWrapper(result);
         TORCH_INTERNAL_ASSERT(wrapper == nullptr);
-        auto* batched = maybeGetBatchedImpl(tensor);
+        auto* batched = maybeGetBatchedImpl(result);
```

Comment:
This can happen if someone invokes a functorch transform on a dead GradTensorWrapper and ends up wrapping the dead GradTensorWrapper in an alive wrapper. Some situations where it can happen are:
- if we do vjp(vmap over an autograd.Function with generate_vmap_rule=True
- if we do vjp( over an autograd.Function with a vmap staticmethod AND the backward pass calls a functorch transform. vjp saves GradTensorWrapper for backward, so in the backward pass, all of those GradTensorWrapper are dead.

This doesn't happen if we do vjp(vmap over a regular PyTorch operator, because the implementation of the operator doesn't end up calling a functorch transform.

Reply:
Makes sense, thanks for clarifying!